hudi-io/src/main/java/org/apache/hudi/storage/HoodieStorage.java

```diff
@@ -267,7 +270,7 @@ public final void createImmutableFileInPath(StoragePath path,
 
       if (content.isPresent() && needTempFile) {
         StoragePath parent = path.getParent();
-        tmpPath = new StoragePath(parent, path.getName() + TMP_PATH_POSTFIX);
+        tmpPath = new StoragePath(parent, path.getName() + "." + UUID.randomUUID());
```

Comment:
A bit worry here, is it possible many callers calling this method to create the same file and leaving many tmp files un cleaned, for example, these callers are suddenly OOM after create these tmp files.

Reply:
The method itself would take care of the file deletion, the original logic also has this concen, and to be worse, if a corrupt tmp file already exists, the file creation would never succeed. That is the best we can do here.

Reply:
I see, make sense